manifests/config.pp

```diff
@@ -171,7 +171,7 @@
   if $force_updatedb and $ensure {
     exec { 'force_updatedb':
       command => $_updatedb_command,
-      creates => "/var/lib/${locate}/${locate}.db",
+      unless  => "/usr/bin/test -s /var/lib/${locate}/${locate}.db",
```

Comment:
Why switch over from `creates`? I'd expect that to work.

Reply:
on Archlinux /var/lib/plocate/plocate.db exist but empty

so force_updatedb not run and on archlinux acceptance fails

The only way I've found was to check with test if it exist and not empty to run the forced update

Reply:
That's weird. I always interpreted `creates` as an existence check, not a non-empty check. The documentation also only talks about that: https://www.puppet.com/docs/puppet/8/type.html#exec-attribute-creates

The code also suggests that it shouldn't care about (non-)empty: https://github.com/puppetlabs/puppet/blob/fce49baa3767fffb000f1b26d3a7b4edbec58999/lib/puppet/type/exec.rb#L420-L450 (which on UNIX translates to `File.exist?(path)`).

In other words, this shouldn't happen.

I wonder what's in the debug output. We don't run our CI with `BEAKER_PUPPET_DEBUG=1` but that should run puppet with `--debug` providing us with the details that may help.

Reply:
Try to better explain

Creates check if the file exist if not run the command

On archlinux file exist so the command never run.

Test -s check is file exist and not empty

So it execute if file not exist  like with creates,  and also if it is empty (like in archlinux)

Reply:
Ah like that. Then it is a behavioral change that should be documented as such.

Reply:
Had a quick play to check if it's possible to generate an empty db file in any way after running updatedb and its impossible since metadata is always saved to the db.

So empty and not existing are equivalent in the context of mlocate - they both indicated that it has  never been run.